src/MongoDB/Cursor.c

```diff
@@ -64,6 +64,11 @@ static int php_phongo_cursor_valid(php_phongo_cursor_t* cursor) /* {{{ */
 
 static void php_phongo_cursor_get_current_key(php_phongo_cursor_t* cursor, zval* key) /* {{{ */
 {
+	if (Z_ISUNDEF(cursor->visitor_data.zchild)) {
+		ZVAL_NULL(key);
+		return;
+	}
+
```

Comment:
Since you introduced the Iterator methods, it looks like `php_phongo_cursor_get_current_key` and `php_phongo_cursor_get_current_data` are only called from `key()` and `current()`, respectively. Can we just delete these two functions?

That would allow you to move this logic into `key()`, which would be consistent with what you did for `current()` in https://github.com/mongodb/mongo-php-driver/commit/70a097e5a145b6eff9931171a86b4c738488bfe8.

Reply:
Sure, I can refactor this 👍